src/brpc/couchbase.cpp

```diff
@@ -0,0 +1,2978 @@
+              channel_, server_address_, selected_bucket_, &request, &response);
+  return result;
+}
+
+CouchbaseOperations::Result CouchbaseOperations::delete_(
+    const string& key, string collection_name) {
+  CouchbaseRequest request(&local_bucket_to_collection_manifest_);
+  CouchbaseResponse response;
+  brpc::Controller cntl;
+  CouchbaseOperations::Result result;
+  if (!sendRequest(CouchbaseOperations::DELETE, key, "", collection_name,
+                   &result, channel_, server_address_, selected_bucket_,
+                   &request, &response)) {
+    return result;
+  }
+  return result;
+}
+
+CouchbaseOperations::Result CouchbaseOperations::add(const string& key,
+                                                     const string& value,
+                                                     string collection_name) {
+  CouchbaseRequest request(&local_bucket_to_collection_manifest_);
+  CouchbaseResponse response;
+  brpc::Controller cntl;
+  CouchbaseOperations::Result result;
+  sendRequest(CouchbaseOperations::ADD, key, value, collection_name, &result,
+              channel_, server_address_, selected_bucket_, &request, &response);
+  return result;
+}
+
+CouchbaseOperations::Result CouchbaseOperations::authenticate(
+    const string& username, const string& password,
+    const string& server_address, const string& bucket_name) {
+  return authenticateAll(username, password, server_address, bucket_name, false,
+                      "");
+}
+
+CouchbaseOperations::Result CouchbaseOperations::authenticateSSL(
+    const string& username, const string& password,
+    const string& server_address, const string& bucket_name,
+    string path_to_cert) {
+  return authenticateAll(username, password, server_address, bucket_name, true,
+                      path_to_cert);
+}
+
+CouchbaseOperations::Result CouchbaseOperations::authenticateAll(
+    const string& username, const string& password,
+    const string& server_address, const string& bucket_name, bool enable_ssl, string path_to_cert) {
+  // Create a channel to the Couchbase server
+  brpc::ChannelOptions options;
+  options.protocol = brpc::PROTOCOL_COUCHBASE;
+  options.connection_type = "single";
+  options.timeout_ms = 1000;  // 1 second
+  options.max_retry = 3;
+  
+  // CRITICAL: Set unique connection_group to prevent connection sharing
+  // Each CouchbaseOperations instance connected to same bucket gets its own connection group
+  options.connection_group = server_address + bucket_name;
+
+  // enable_ssl
+  if (enable_ssl) {
+    brpc::ChannelSSLOptions* ssl_options = options.mutable_ssl_options();
+    ssl_options->sni_name = server_address;
+    ssl_options->verify.verify_depth =
+        1;  // Enable certificate verification, to disable SSL set it to 0
+    ssl_options->verify.ca_file_path =
+        path_to_cert;  // Path to your downloaded TLS certificate
+  }
+  CouchbaseOperations::Result result;
+  brpc::Channel* new_channel = new brpc::Channel();
```

Comment:
It seems that you are creating `brpc::Channel` inside the `CouchbaseOperations` implementation. This is not a common design of brpc. You can see other protocols' usage, such as the Memcached or Redis protocols: The Channel is created and initialized by user, and user can control specific ChannelOptions. This is one of the most useful features of brpc.
I think you can follow this convention, you can accept user created channels, such as `CouchbaseOperations::authenticate(Channel& channel, const string& username, const string& password, const string& bucket_name)`. Such interface may be a little more complex to use than your current one, but it can allow use to leverage the full functions of `brpc::Channel`.